src/evict/evict_lru.c

```diff
@@ -1720,10 +1720,22 @@ __evict_walk_file(WT_SESSION_IMPL *session,
 			 * Try a different walk start point next time if a
 			 * walk gave up.
 			 */
-			btree->evict_walk_state =
```

Comment:
> since it meant we didn't need to update another section of code when adding a new walk state

Yes, but it also means the enum's *order of declaration* is part of the switching algorithm, which is just weird. :)

It's your call. If you hate the change, revert it and I'll either add in the casts or ignore the complaints.